xt_training/runner.py

```diff
@@ -145,10 +176,10 @@ def __call__(self, loader, mode=None, return_preds=False):
                 logger(loss, metrics, i_batch)
 
                 if return_preds:
-                    y_pred_epoch.append(y_pred.detach().cpu())
-                    y_epoch.append(y.detach().cpu())
+                    y_pred_epoch.append(detach_objects(y_pred))
+                    y_epoch.append(detach_objects(y))
 
-        if model.training and scheduler is not None:
+        if model.training and scheduler is not None and not is_batch_scheduler:
             scheduler.step()
             self.epoch += 1
```

Comment:
This will break the self.epoch counter. I think it should be:
```python
if model.training:
    if not is_batch_scheduler and scheduler is not None:
        scheduler.step()
    self.epoch += 1
```

(This was already broken by me the way it was, don't feel bad :))